src/transforms/filter.js

```diff
@@ -54,6 +63,16 @@ exports.supplyDefaults = function(transformIn, fullData, layout) {
     coerce('value');
     coerce('filtersrc');
 
+    // numeric values as character should be converted to numeric
+    if(Array.isArray(transformOut.value)) {
+        transformOut.value = transformOut.value.map(function(v) {
+            if(isNumeric(v)) v = +v;
+            return v;
+        });
+    } else {
+        if(isNumeric(transformOut.value)) transformOut.value = +transformOut.value;
```

Comment:
FYI I haven't touched anything in `filter.js` so it's now as it is in @timelyportfolio's branch (and I haven't yet pushed the changes to `groupby.js`.  If I understand the task correctly, I need to work on `groupby` as part of the first sprint on it so I'll get back on it in a bit.